zap/src/main/java/org/zaproxy/zap/model/Vulnerabilities.java

```diff
@@ -93,8 +54,16 @@ private static synchronized void init() {
      *     {@code null}.
      */
     public static List<Vulnerability> getAllVulnerabilities() {
-        initializeIfEmpty();
-        return vulnerabilities;
+        var local = provider;
+        if (local == null) {
+            logNoProvider();
+            return List.of();
+        }
+        return local.getAll();
+    }
+
+    private static void logNoProvider() {
+        LOGGER.error("No provider found.", new Exception());
```

Comment:
Should the new Exception have some details? If not why not just skip it? Should the literal string say "vulnerability data provider"?

Reply:
The exception is just to create a stack trace (for debug purposes). The logged class name already indicates it's for vulnerabilities.

Reply:
Ok